src/cljs/rems/application.cljs

```diff
@@ -125,8 +129,18 @@
                             {:form (:form/id form)
                              :field (:field/id field)
                              :value (:field/value field)})
-                          (build-index {:keys [:form :field] :value-fn :value}))]
-    (assoc db ::edit-application {:field-values field-values
+                          (build-index {:keys [:form :field] :value-fn :value}))
+        duo-codes (->> (for [duo (-> application :application/duo :duo/codes)]
+                         (update-existing duo
+                                          :restrictions
+                                          (partial build-index {:keys [:type]
+                                                                :value-fn (fn [{:keys [type values]}]
+                                                                            (case type
+                                                                              :mondo values
+                                                                              (-> values first :value)))})))
```

Comment:
Why do we need to process the codes here? Couldn't the application contain all the relevant info when it is loaded without the need to do extra fetching or processing?

Reply:
i'm not sure what you mean - this block transforms application duo codes into a form that is usable by the application editor. in this case restriction data is indexed so that it is easier to refer to by front end components.

Reply:
Ok, I guess it's fine. Just looked like a big expression needed to do it.

Reply:
extracted the main gist into separate function. i guess it looks more neat now.